src/proof/lfsc/lfsc_printer.cpp

```diff
@@ -0,0 +1,515 @@
+    case PfRule::TRUE_INTRO:
+    case PfRule::FALSE_INTRO:
+    case PfRule::TRUE_ELIM:
+    case PfRule::FALSE_ELIM: pf << h << cs[0]; break;
+    // arithmetic
+    case PfRule::ARITH_MULT_POS:
+    case PfRule::ARITH_MULT_NEG:
+    {
+      // do not pass type (as[0].getType())
+      pf << h << as[0] << as[1];
+    }
+    break;
+    case PfRule::ARITH_TRICHOTOMY:
+    {
+      // should be robust to different orderings
+      pf << h << h << h << cs[0] << cs[1];
+    }
+    break;
+    // strings
+    case PfRule::STRING_LENGTH_POS: pf << as[0]; break;
+    case PfRule::STRING_LENGTH_NON_EMPTY: pf << h << cs[0]; break;
+    case PfRule::RE_INTER: pf << h << h << h << cs[0] << cs[1]; break;
+    case PfRule::CONCAT_EQ:
+      pf << h << h << h << args[0].getConst<bool>() << cs[0];
+      break;
+    case PfRule::CONCAT_UNIFY:
+      pf << h << h << h << h << args[0].getConst<bool>() << cs[0] << cs[1];
+      break;
+    case PfRule::CONCAT_CSPLIT:
+      pf << h << h << h << h << args[0].getConst<bool>() << cs[0] << cs[1];
+      break;
+      break;
+    case PfRule::RE_UNFOLD_POS:
+      if (children[0]->getResult()[1].getKind() != REGEXP_CONCAT)
+      {
+        return false;
+      }
+      pf << h << h << h << cs[0];
+      break;
+    case PfRule::STRING_EAGER_REDUCTION:
+    {
+      Kind k = as[0].getKind();
+      if (k == STRING_TO_CODE || k == STRING_CONTAINS || k == STRING_INDEXOF)
+      {
+        pf << h << as[0] << as[0][0].getType();
+      }
+      else
+      {
+        // not yet defined for other kinds
+        return false;
+      }
+    }
+    break;
+    case PfRule::STRING_REDUCTION:
+    {
+      Kind k = as[0].getKind();
+      if (k == STRING_SUBSTR || k == STRING_INDEXOF)
+      {
+        pf << h << as[0] << as[0][0].getType();
+      }
+      else
+      {
+        // not yet defined for other kinds
+        return false;
+      }
+    }
+    break;
+    // quantifiers
+    case PfRule::SKOLEM_INTRO:
+    {
+      pf << h << d_tproc.convert(SkolemManager::getOriginalForm(args[0]));
+    }
+    break;
+    // ---------- arguments of non-translated rules go here
+    case PfRule::LFSC_RULE:
+    {
+      LfscRule lr = getLfscRule(args[0]);
+      // lambda should be processed elsewhere
+      Assert(lr != LfscRule::LAMBDA);
+      // Note that `args` has 2 builtin arguments, thus the first real argument
+      // begins at index 2
+      switch (lr)
+      {
+        case LfscRule::SCOPE: pf << h << as[2] << cs[0]; break;
+        case LfscRule::NEG_SYMM: pf << h << h << cs[0]; break;
+        case LfscRule::CONG: pf << h << h << h << h << cs[0] << cs[1]; break;
+        case LfscRule::AND_INTRO1: pf << h << cs[0]; break;
+        case LfscRule::NOT_AND_REV: pf << h << h << cs[0]; break;
+        case LfscRule::PROCESS_SCOPE: pf << h << h << as[2] << cs[0]; break;
+        case LfscRule::AND_INTRO2: pf << h << h << cs[0] << cs[1]; break;
+        // do not pass type (cs[0]->getResult()[0].getType())
```

Comment:
Can this be made an `Assert`?

Reply:
This is leftover from when i added subtypes to the LFSC signature, this comment is no longer necessary.